### ISSUE TITLE
Common Loss Functions written natively in JAX?

### ISSUE THREAD

**Issue by derrickxli** [google/jax#7175]
Could we create a new subpackage (or find somewhere to put this) for native loss functions written in JAX? The issue is that when I try to use loss functions from say `sk_metrics`, the function will not be able to take advantage of autodiff from `jax.grad` because of the incompatibility of tracing device arrays through vanilla numpy functions.

If there were a suite of common native written loss functions that could use `jax.loss`, then we could have the ability to use loss functions written from a standardized, robust library while also taking advantage of autodiff. This way, users will not create their own error prone loss functions.

**Comment by jakevdp** [comment]
Thanks for the question! I don't know of any centralized effort at that yet, but one example where something like this exists is within the optax project: https://github.com/deepmind/optax/blob/master/optax/_src/loss.py

**Comment by mblondel** [comment]
We also have some here https://github.com/google/jaxopt/blob/main/jaxopt/_src/loss.py

**Comment by froystig** [comment]
As @tomhennigan put it in https://github.com/google/jax/pull/7575#issuecomment-896556986:

> some folks in the community have discussed this before, and felt that losses should be provided by libraries on top of JAX rather than be a part of JAX itself (to keep JAX focussed and lightweight).

There are loss definitions available in optax, jaxopt, RLAX, and other tested and maintained libraries, all of which are compatible with JAX's autodiff transformations.

I'll close this issue since it is out of jax's scope for the time being. We can still continue the discussion, of course.

**Comment by mwitiderrick** [comment]
Check this JAX loss functions article https://www.machinelearningnuggets.com/jax-loss-functions/
